src/test/integration/IntegrationBase.t.sol

```diff
@@ -477,20 +478,83 @@ abstract contract IntegrationBase is IntegrationDeployer {
         strategies[0] = strat;
         removedShares[0] = _removedShares;
 
-        assert_Snap_Removed_StakerShares(staker, strategies, removedShares, err);
+        assert_Snap_Removed_StakerDepositShares(staker, strategies, removedShares, err);
+    }
+
+    /// @dev Check that the staker's withdrawable shares have decreased by `removedShares`
+    function assert_Snap_Removed_StakerWithdrawableShares(
+        User staker, 
+        IStrategy[] memory strategies, 
+        uint[] memory removedShares,
+        string memory err
+    ) internal {
+        uint[] memory curShares = _getStakerWithdrawableShares(staker, strategies);
+        // Use timewarp to get previous staker shares
+        uint[] memory prevShares = _getPrevStakerWithdrawableShares(staker, strategies);
+
+        // For each strategy, check (prev - removed == cur)
+        for (uint i = 0; i < strategies.length; i++) {
+            assertEq(prevShares[i] - removedShares[i], curShares[i], err);
+        }
+    }
+
+    function assert_Snap_Removed_StakerWithdrawableShares(
+        User staker, 
+        IStrategy strat, 
+        uint _removedShares,
+        string memory err
+    ) internal {
+        IStrategy[] memory strategies = new IStrategy[](1);
+        uint[] memory removedShares = new uint[](1);
+        strategies[0] = strat;
+        removedShares[0] = _removedShares;
+
+        assert_Snap_Removed_StakerWithdrawableShares(staker, strategies, removedShares, err);
+    }
+
+    /// @dev Check that the staker's withdrawable shares have decreased by at least `removedShares`
+    /// @dev Used to handle overslashing of beacon chain
+    function assert_Snap_Removed_Staker_WithdrawableShares_AtLeast(
+        User staker,
+        IStrategy[] memory strategies,
+        uint[] memory removedShares,
+        string memory err
+    ) internal {
+        uint[] memory curShares = _getStakerWithdrawableShares(staker, strategies);
+        // Use timewarp to get previous staker shares
+        uint[] memory prevShares = _getPrevStakerWithdrawableShares(staker, strategies);
+
+        // For each strategy, check (prev - removed == cur)
+        for (uint i = 0; i < strategies.length; i++) {
+            assertGe(prevShares[i] - removedShares[i], curShares[i], err);
```

Comment:
Can potentially use something like [assertApproxEqAbs](https://book.getfoundry.sh/reference/ds-test?highlight=greater#assertapproxeqabs) or [assertApproxEqRel](https://book.getfoundry.sh/reference/ds-test?highlight=greater#assertapproxeqrel)

Reply:
Added to be 1 GWEI for now

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/857/commits/d6c5281ff3fc7971d4f493d565d6b379a862ff70